src/Collection.php

```diff
@@ -244,34 +250,84 @@ public function count(array $filter = array(), array $options = array())
-     * @param array|object $keys
-     * @param array        $options
+     * @see Collection::createIndexes()
+     * @param array|object $key     Document containing fields mapped to values,
+     *                              which denote order or an index type
+     * @param array        $options Index options
      * @return string The name of the created index
      */
-    public function createIndex($keys, array $options = array())
+    public function createIndex($key, array $options = array())
     {
-        // TODO
+        return current($this->createIndexes(array(array('key' => $key) + $options)));
     }
 
     /**
-     * Create multiple indexes in the collection.
+     * Create one or more indexes for the collection.
+     *
+     * Each element in the $indexes array must have a "key" document, which
+     * contains fields mapped to an order or type. Other options may follow.
+     * For example:
      *
-     * TODO: decide if $models should be an array of associative arrays, using
-     * createIndex()'s parameter names as keys, or tuples, using parameters in
-     * order (e.g. [keys, options]).
+     *     $indexes = [
+     *         // Create a unique index on the "username" field
+     *         [ 'key' => [ 'username' => 1 ], 'unique' => true ],
+     *         // Create a 2dsphere index on the "loc" field with a custom name
+     *         [ 'key' => [ 'loc' => '2dsphere' ], 'name' => 'geo' ],
+     *     ];
+     *
+     * If the "name" option is unspecified, a name will be generated from the
+     * "key" document.
      *
      * @see http://docs.mongodb.org/manual/reference/command/createIndexes/
      * @see http://docs.mongodb.org/manual/reference/method/db.collection.createIndex/
-     * @param array $models
+     * @param array $indexes List of index specifications
      * @return string[] The names of the created indexes
+     * @throws InvalidArgumentException if an index specification does not
+     *                                  contain a "key" document
+     * @throws UnexpectedTypeException if an index specification is not an array
+     *                                 or a "key" document is not an array or
+     *                                 object
      */
-    public function createIndexes(array $models)
+    public function createIndexes(array $indexes)
     {
-        // TODO
+        foreach ($indexes as &$index) {
+            if ( ! is_array($index)) {
+                throw new UnexpectedTypeException($index, 'array');
+            }
+
+            if ( ! isset($index['key'])) {
+                throw new InvalidArgumentException('Required "key" document is missing from index specification');
+            }
+
+            if ( ! is_array($index['key']) && ! is_object($index['key'])) {
+                throw new UnexpectedTypeException($index['key'], 'array or object');
+            }
+
+            $index['key'] = (object) $index['key'];
+            $index['ns'] = $this->ns;
```

Comment:
If the user specifies `$index['ns']`, we should cast it to a string. We can let the user provide incorrect strings here, and just set the `$this->ns` default if the option is not provided.